pdl-compiler/src/analyzer.rs

```diff
@@ -1618,6 +1621,98 @@ fn check_optional_fields(file: &File) -> Result<(), Diagnostics> {
     diagnostics.err_or(())
 }
 
+/// Check field offsets.
+/// Raises error diagnostics for the following cases:
+///      - non bit-field field not aligned to a octet boundary
+fn check_field_offsets(file: &File, scope: &Scope, schema: &Schema) -> Result<(), Diagnostics> {
+    let mut diagnostics: Diagnostics = Default::default();
+    for decl in &file.declarations {
+        let mut offset = 0;
+
+        for field in decl.fields() {
+            match &field.desc {
+                FieldDesc::Typedef { type_id, .. }
+                    if matches!(
+                        scope.typedef.get(type_id),
+                        Some(Decl { desc: DeclDesc::Enum { .. }, .. })
+                    ) => {}
+                FieldDesc::Payload { .. }
+                | FieldDesc::Body { .. }
+                | FieldDesc::Typedef { .. }
+                | FieldDesc::Array { .. }
+                | FieldDesc::Padding { .. }
+                | FieldDesc::Checksum { .. } => {
+                    if offset % 8 != 0 {
+                        diagnostics.push(
+                            Diagnostic::error()
+                                .with_code(ErrorCode::InvalidFieldOffset)
+                                .with_message(format!(
+                                    "{} field is not aligned to an octet boundary",
```

Comment:
Would it perhaps make sense to report `offset % 8` in the error message, or is that normally obvious when looking at the fields?

Reply:
I am afraid that this value would be confusing in the error message in case of more complicated messages :
the offset is reset here after aligned fields, the following packet would display `1` as offset:

```
packet Test {
  _size_(foo) : 8,
  foo: 8[],
  _reserved_ : 1,
  bar: 8[],
}
```